File: UIWebView离线缓存/oc/NSURLProtocolOC-webView/NSURLProtocolOC-webView/MHCacheTool/MHCache.m
//
//  MHCache.m
//  NSURLProtocolOC-webView
//
//  Created by 胡明昊 on 16/12/26.
//  Copyright © 2016年 CMCC. All rights reserved.
//

#import "MHCache.h"

static NSString *const cacheName = @"MHCacheName";

@implementation MHCache

+ (instancetype)defaultcache {
    
    static MHCache *cache = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        cache = [[MHCache alloc] initWithName:cacheName];
    });
    
    return cache;
}

@end

File: UIWebView离线缓存/swift/NSURLProtocol-webView/NSURLProtocol-webView/LoadCacheTool/MHCache.swift
//
//  MHCache.swift
//  NSURLProtocol-webView
//
//  Created by 胡明昊 on 16/12/26.
//  Copyright © 2016年 CMCC. All rights reserved.
//

import UIKit

private let CacheName = "MHCacheName"
/// 单例
class MHCache: YYCache {
    
    static var shareInstance: MHCache {
        struct MyStatic {
            static var instance: MHCache = MHCache(name: CacheName)!
        }
        
        return MyStatic.instance
    }
    
}
